### ISSUE TITLE
What's the proper way to implement chatting feature?

### ISSUE THREAD

**Issue by finom** [Stability-AI/StableLM#67]
Hey guys! To run the basic example I can use labels like "<|SYSTEM|>", "<|USER|>", "<|ASSISTANT|>". That's clear how to run the example script and how to ask one question but I don't really understand how can I turn that into a chat so the LM has understanding in what has been discussed before. As far as I understand I need to concat all the conversation like 

```py
"""
<|SYSTEM|>...
<|USER|>How are you?
<|ASSISTANT|>Good
<|USER|>In other words?
<|ASSISTANT|>Amazing
"""
```

and pass it to `tokenizer` as a prompt. When a new message appears I just append it to the end of the prompt. Is that correct?

**Comment by RomboutKarelse** [comment]
Did you see this? https://github.com/Stability-AI/StableLM/blob/main/notebooks/stablelm-alpha.ipynb

**Comment by finom** [comment]
@RomboutKarelse @mcmonkey4eva  as far as I understand I need to send the entire chat history as I'd do with OpenAI API?

**Comment by mcmonkey4eva** [comment]
Yes correct. The LLM doesn't have any internal memory beyond what you feed into it.

**Comment by RomboutKarelse** [comment]
Ah, I should have pointed you to this code. https://huggingface.co/spaces/stabilityai/stablelm-tuned-alpha-chat/blob/main/app.py
This implements a chatbot with history.

**Comment by finom** [comment]
@RomboutKarelse @mcmonkey4eva thank you guys!
